patternlab/styleguide/source/_patterns/02-molecules/main-nav.twig

```diff
@@ -1,21 +1,21 @@
 <div class="ma__main-nav">
-  <ul class="ma__main-nav__items js-main-nav">
+  <ul class="ma__main-nav__items js-main-nav" role="menubar">
     {% for nav in mainNav %}
       {% set buttonId =  'button' ~ loop.index %}
       {% set menuId =  'menu' ~ loop.index %}
-      <li class="ma__main-nav__item {{ nav.active ? 'is-active' : '' }} {{ nav.subNav ? 'has-subnav js-main-nav-toggle' : 'js-main-nav-top-link' }}">
+      <li  role="menuitem" class="ma__main-nav__item {{ nav.active ? 'is-active' : '' }} {{ nav.subNav ? 'has-subnav js-main-nav-toggle' : 'js-main-nav-top-link' }}" tabindex="-1">
         {% if nav.subNav %}
-          <button id="{{ buttonId }}" class="ma__main-nav__top-link" aria-haspopup="true" aria-expanded="false" tabindex="{{ loop.first? 0 : -1 }}">{{ nav.text }}</button>
+          <button id="{{ buttonId }}" class="ma__main-nav__top-link" aria-haspopup="true" aria-expanded="false" tabindex="0"><span class="visually-hidden show-label">Show the sub topics of </span>{{ nav.text }}</button>
           <div class="ma__main-nav__subitems js-main-nav-content is-closed">
             <ul id="{{ menuId }}" role="menu" aria-labelledby="{{ buttonId }}" class="ma__main-nav__container">
-              <li role="menuitem" class="ma__main-nav__subitem"><a href="{{ nav.href }}" class="ma__main-nav__link" tabindex="-1">{{ nav.text }}</a></li>
+              <li role="menuitem" class="ma__main-nav__subitem"><a href="{{ nav.href }}" class="ma__main-nav__link">{{ nav.text }}</a></li>
               {% for subNav in nav.subNav %}
-                <li role="menuitem" class="ma__main-nav__subitem"><a href="{{ subNav.href }}" class="ma__main-nav__link" tabindex="-1">{{ subNav.text }}</a></li>
+                <li class="ma__main-nav__subitem"><a role="menuitem" href="{{ subNav.href }}" class="ma__main-nav__link">{{ subNav.text }}</a></li>
               {% endfor %}
-              <li role="menuitem" class="ma__main-nav__subitem">
-                <a href="{{ nav.href }}" class="ma__main-nav__link" tabindex="-1">
+              <li class="ma__main-nav__subitem">
+                <a role="menuitem" href="{{ nav.href }}" class="ma__main-nav__link">

```

Comment:
Please move the `role="menuitem"` from `<a>` to `<li>` to maintain the link role on `<a>`.